tests/imaging/test_lattice.py

```diff
@@ -0,0 +1,1382 @@
+
+    def test_lattice_serialization_methods_exist(self, simple_lattice):
+        """Test that serialization methods exist (if applicable)."""
+        # AutoSerialize typically provides to_dict, from_dict, etc.
+        # Check if these methods are available
+        if hasattr(simple_lattice, "to_dict"):
+            assert callable(getattr(simple_lattice, "to_dict"))
+        if hasattr(simple_lattice, "from_dict"):
+            assert callable(getattr(simple_lattice, "from_dict"))
+
+
+class TestLatticeAttributes:
+    """Test internal attributes and state management."""
+
+    @pytest.fixture
+    def lattice_with_state(self):
+        """Create lattice with some state."""
+        image = np.random.randn(100, 100)
+        lattice = Lattice.from_data(image)
+
+        # Mock lattice parameters
+        lattice._lat = np.array([[10.0, 10.0], [10.0, 0.0], [0.0, 10.0]])
+
+        return lattice
+
+    def test_lattice_has_lat_attribute(self, lattice_with_state):
+        """Test that lattice has _lat attribute after fitting."""
+        assert hasattr(lattice_with_state, "_lat")
+        assert isinstance(lattice_with_state._lat, np.ndarray)
+
+    def test_lattice_lat_shape(self, lattice_with_state):
+        """Test that _lat has correct shape (3, 2)."""
+        assert lattice_with_state._lat.shape == (3, 2)
+
+    def test_lattice_lat_components(self, lattice_with_state):
+        """Test that _lat contains origin, u, and v vectors."""
+        r0, u, v = lattice_with_state._lat
+
+        assert r0.shape == (2,)
+        assert u.shape == (2,)
+        assert v.shape == (2,)
+
+    def test_lattice_image_is_dataset2d(self):
+        """Test that internal image is always Dataset2d."""
+        image = np.random.randn(100, 100)
+        lattice = Lattice.from_data(image)
+
+        assert isinstance(lattice._image, Dataset2d)
+
+
+class TestLatticeRobustness:
+    """Test robustness to various inputs and conditions."""
+
+    def test_lattice_with_single_pixel(self):
+        """Test lattice with 1x1 image."""
+        image = np.array([[1.0]])
+
+        lattice = Lattice.from_data(image)
+
+        assert lattice.image.shape == (1, 1)
+
+    def test_lattice_with_single_row(self):
+        """Test lattice with single row."""
+        image = np.random.randn(1, 100)
+
+        lattice = Lattice.from_data(image)
+
+        assert lattice.image.shape == (1, 100)
+
+    def test_lattice_with_single_column(self):
+        """Test lattice with single column."""
+        image = np.random.randn(100, 1)
+
+        lattice = Lattice.from_data(image)
+
+        assert lattice.image.shape == (100, 1)
+
+    def test_lattice_with_bool_array(self):
+        """Test lattice creation with boolean array."""
+        image = np.random.rand(50, 50) > 0.5
+
+        lattice = Lattice.from_data(image)
+
+        assert lattice is not None
+
+    def test_lattice_with_complex_numbers(self):
```

Comment:
Why are we supporting complex numbers here?  Raises a warning in pytest too:

tests/imaging/test_lattice.py::TestLatticeRobustness::test_lattice_with_complex_numbers
  /Users/cophus/repos/quantem/src/quantem/imaging/lattice.py:69: ComplexWarning: Casting complex values to real discards the imaginary part
    arr = arr.astype(float, copy=False)